app/models/repository.rb

```diff
@@ -206,8 +206,19 @@ def self.create_or_update!(repo)
         digest = ""
       end
 
-      Tag.create!(name: tag, repository: repository, author: portus, digest: digest, image_id: id)
-      logger.tagged("catalog") { logger.info "Created the tag '#{tag}'." }
+      if current_tag = repository.tags.find_by_name(tag)
```

Comment:
Why? I mean, we were using the `to_be_created_tags` array because it contained tags that we were sure didn't exist, therefore there is nothing to update, only to create. Do you have a use case in which this is not true ? As in, maybe we've got a bug on the `to_be_created_tags`, is that what you are saying ? :smile: